components/script/dom/transitionevent.rs

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::codegen::Bindings::EventBinding::EventMethods;
+use dom::bindings::codegen::Bindings::TransitionEventBinding;
+use dom::bindings::codegen::Bindings::TransitionEventBinding::{TransitionEventInit, TransitionEventMethods};
+use dom::bindings::error::Fallible;
+use dom::bindings::global::GlobalRef;
+use dom::bindings::inheritance::Castable;
+use dom::bindings::js::Root;
+use dom::bindings::num::Finite;
+use dom::bindings::reflector::reflect_dom_object;
+use dom::bindings::str::DOMString;
+use dom::event::Event;
+use string_cache::Atom;
+
+#[dom_struct]
+pub struct TransitionEvent {
+    event: Event,
+    property_name: Atom,
+    elapsed_time: Finite<f32>,
+    pseudo_element: DOMString,
+}
+
+impl TransitionEvent {
+    #[allow(unrooted_must_root)]
+    pub fn new_inherited(type_: Atom, init: &TransitionEventInit) -> TransitionEvent {
+        let event = Event::new_inherited();
+        event.init_event(type_, init.parent.bubbles, init.parent.cancelable);
```

Comment:
nit: In the other event structs the `new_inherited` usually doesn't handle calling `init_event`, this is typically done in `fn new()`. This would avoid `#[allow(unrooted_must_root)]`

see https://dxr.mozilla.org/servo/source/components/script/dom/closeevent.rs#26-33,41-58